test/unit/math/prim/fun/grad_2F1_test.cpp

```diff
@@ -129,3 +129,19 @@ TEST(MathPrimScalFun, grad2F1_10) {
   EXPECT_NEAR(0.4617734323582945, grad_a1, 1e-8);
   EXPECT_NEAR(-0.4617734323582945, grad_b1, 1e-8);
 }
+
+TEST(MathPrimScalFun, grad2F1_11) {
+  double a1 = 3.70975;
+  double a2 = 1.0;
+  double b1 = 2.70975;
+  double z = 0.999696;
+
+  double grad_a1;
+  double grad_b1;
+  stan::math::grad_2F1(grad_a1, grad_b1, a1, a2, b1, z);
+
+  EXPECT_NEAR(29369830.002773938200417693317785, grad_a1,
+              1e1);  // reference: discrete diff in mathematica
```

Comment:
Isn't 10 a big error here? That's like a relative error of `3e-7`

Reply:
Right, but these have *never* been better as far as I know.  This section of the function's range should really be using a different calculation (the gradients just become more and more extreme as `z` approaches 1).  I'd enjoy re-working these to be more complete but I'm not sure it's worth the trade-off unless we show there's some inaccuracy in a model that's based on this issue.

Reply:
Makes sense